docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+
+```rust
+impl RequestBuilder<Event> for BasicRequestBuilder {
+```
+
+There are a number of stages in the request builder process -  first the input 
+is split out into metadata and actual event data. The event data is encoded. 
+The results from the encoding are are passed with the metadata to create the 
+final request that is passed to the `Tower` sink.
+
+Here, the trait is generic over `Event` which is the input type that is 
+passed in to start the request building process.
+
+There are a number of associated types:
+
+```rust
+    type Metadata = EventFinalizers;
+    type Events = Event;
+    type Encoder = BasicEncoder;
+    type Payload = Bytes;
+    type Request = BasicRequest;
+    type Error = std::io::Error;
+```
+
+## Metadata
+
+`Metadata` is the type for any information to be passed while building the request
+that is additional to the actual event being used. In this case we just need the
+`EventFinalizers`.
+
+## Events
+
+Events contains the type that is the event type passed to the `Encoder`.
+
+## Encoder
+
+The `Encoder` is the type that is used to encode the event to create the final
+payload. We are using the `BasicEncoder` described earlier.
+
+## Payload
+
+The `Payload` is the type of the final data that is encoded.
+
+## Request
+
+`Request` is the type that is sent to the final service. This is the `BasicRequest`
+we described earlier.
+
+## Error
+
+`Error` is the type of any errors that are creating while encoding the event.
+
+
+The following functions for the `RequestBuilder` trait need implementing:
+
+## compression
+
+```rust
+    fn compression(&self) -> Compression {
+        Compression::None
+    }
+```
+
+The payload for the built request can be compressed. Here we return `Compression::None`
+to indicate that we will not be compressing. `Compression::Gzip` and `Compression::Zlib`
+are available choices.
+
+
+## encoder
+
+```rust
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoder
+    }
+```
+
+We return the encoder to use. This is the `BasicEncoder` defined earlier.
+
+
+## split_input
+
+```rust
+    fn split_input(
+        &self,
+        mut input: Event,
+    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
+        let finalizers = input.take_finalizers();
+        let metadata_builder = RequestMetadataBuilder::from_events(&input);
+        (finalizers, metadata_builder, input)
+    }
+```
+
+`split_input` takes the input and extracts the metadata from the events. In this
+case we are returning the `input` parameter unprocessed. 
+
+This may not always be the case. For example, the `amqp` sink will initially
+process  the event to extract fields to be used to calculate the exchange
+to send the message  to. This is sent  with the event to `split_input`.
+`split_input` splits that out into the event for  encoding and the metadata
+containing the exchange which will be used when sending  the event to `amqp`.
```

Comment:
```suggestion
This may not always be the case. For example, the `amqp` sink will initially
process the event to extract fields to be used to calculate the exchange
to send the message to. This is sent with the event to `split_input`.
`split_input` splits that out into the event for encoding and the metadata
containing the exchange which will be used when sending the event to `amqp`.
```